### ISSUE TITLE
Extend action to support Ubuntu runner?

### ISSUE THREAD

**Issue by stweil** [msys2/setup-msys2#293]
It might sound strange, but the msys packages work nice for cross builds (targetting 32 or 64 bit Windows) on Debian / Ubuntu hosts, too. Typically a cross build on Linux is much faster than a build on Windows. Therefore it would be great if the GitHub action could be used with the Ubuntu 22.04 runner, too.

**Comment by lazka** [comment]
MSYS2 doesn't support running on Ubuntu, so this is nothing the github action can provide.

**Comment by stweil** [comment]
I'd expect that the action provides pacman and pacman settings, so it is possible to install msys packages for mingw32, mingw64, ...

**Comment by lazka** [comment]
I see. The install scripts and hooks would require something like wine though, or you somehow patch them out.

**Comment by stweil** [comment]
I simply ignore the error messages from the install hooks. Typically that works fine.

**Comment by MehdiChinoune** [comment]
> Typically a cross build on Linux is much faster than a build on Windows.

Are you using autotools, then that's expected. Use CMake or meson for a better buildsystem and faster build-time.

**Comment by lazka** [comment]
I could see someone providing a docker image based on Ubuntu as a first step (or Fedora, since they have a ucrt cross compiler toolchain)

**Comment by stweil** [comment]
Yes, Fedora has a more complete selection of library packages for cross builds.
